proxy/http2/Http2ConnectionState.cc

```diff
@@ -1009,6 +1016,19 @@ Http2ConnectionState::main_event_handler(int event, void *edata)
       break;
     }
 
+    // Don't allow anything that might change the state of the server if it is read from early data.
+    // SETTINGS frame is only allowed to be an ACK, which is checked in rcv_settings_frame().
+    // PING frame is ok.
+    // HEADERS frame can only be checked after parsing the payload, need to check the method.
+    // CONTINUATION frame is just extra payload of HEADERS frame.
+    if (frame->is_from_early_data() && frame->header().type != HTTP2_FRAME_TYPE_HEADERS &&
```

Comment:
What about PRIORITY frame? nghttp client sends SETTINGS -> PRIORITY -> HEADERS, I'm not sure if the client supports Early Data though.

Reply:
Yeah, it's kind of tricky with which frames we allow, maybe we'll have to test it in the fields since I can't find any clients that does h2 and early data together, I had to use `openssl s_client` and get the raw hex output and use a decoder I wrote to interpret the output. And the early data I sent during testing is also just PREFACE->SETTINGS (ACK)->HEADERS, since that is the minimum requirement for a h2 connection.

Reply:
Changed it up a little bit.